backend/addresses/views.py

```diff
@@ -1,3 +1,41 @@
-from django.shortcuts import render
+from rest_framework import viewsets, permissions, status
+from rest_framework.decorators import action
+from rest_framework.response import Response
+from .models import Address
+from .serializers import AddressSerializer
+from accounts.models import CustomerProfile
 
-# Create your views here.
+class AddressViewSet(viewsets.ModelViewSet):
+    serializer_class = AddressSerializer
+    permission_classes = [permissions.IsAuthenticated]
+
+    def get_queryset(self):
+        return Address.objects.filter(user=self.request.user)
+
+    def perform_create(self, serializer):
+        # The serializer handles setting the user and is_default logic
+        serializer.save()
+
+    @action(detail=True, methods=['post'], url_path='set-default')
```

Comment:
This `@action` decorator will need to be updated, after a slight change that affects different aspects of our code and business logic:
  - The API Endpoint will change from `POST /api/addresses/{id}/set-default/` into `PATCH /api/addresses/{id}/set-default/`
     - This ensures that we update the `default_shipping_address_id` specifically for a `"customer"`, and not any other field.
     - Makes a clear and distinct separation of concerns from other API endpoints.
     - Triggers the serializer's `update()` method instead of `create()` method, which makes sense in terms of business logic defined there.
     
Kindly, find the following suggestion:
```suggestion
    @action(detail=True, methods=['patch'], url_path='set-default')
```

This means that a request to the `` endpoint, would take something like this in its body:
```json
  {
    "is_default": true
  }
```

Reply:
Dear @mohamed952741, the `post` verb needs to be addressed.
Just replace the `post` verb with `patch`.